src/main/kotlin/org/dashevo/dapiclient/DapiClient.kt

```diff
@@ -1,219 +1,398 @@
-        dapiService.getUser(JsonRPCRequest("getUser", mapOf("username" to username))).enqueue(fun(it: Response<JsonRPCResponse<BlockchainUser>>) {
-            val body = it.body()
-            if (it.isSuccessful && body != null) {
-                if (body.error != null) {
-                    cb.onError(body.error["message"] as String)
-                } else {
-                    cb.onSuccess(body)
-                }
-            } else {
-                cb.onError(it.message())
-            }
-        }, {
-            cb.onError(it.localizedMessage)
-        })
+    /* Core */
+    /** get status of platform node  */
+    fun getStatus(): GetStatusResponse? {
+        logger.log(Level.INFO, "getStatus")
+        val request = CoreOuterClass.GetStatusRequest.newBuilder().build()
+
+        val service = getCoreService()
+
+        val response: CoreOuterClass.GetStatusResponse =  try {
+            service.getStatus(request)
+        } catch (e: StatusRuntimeException) {
+            logger.log(Level.WARNING, "RPC failed: {0}", e.status)
+            throw e
+        } finally {
+            shutdownInternal()
+        }
+
+        return GetStatusResponse(response.coreVersion,
+                response.protocolVersion,
+                response.blocks,
+                response.timeOffset,
+                response.connections,
+                response.proxy,
+                response.difficulty,
+                response.testnet,
+                response.relayFee,
+                response.errors,
+                response.network)
+    }
+
+    fun getBlockByHeight(height: Int): ByteArray? {
+        Preconditions.checkArgument(height > 0)
+        val request = CoreOuterClass.GetBlockRequest.newBuilder()
+                .setHeight(height)
+                .build()
+        return getBlock(request)
     }
 
     /**
-     * Fetch DAP Contract.
-     * @param contractId: DAP ID of registered contract.
-     * @param cb: Callback.
+     *
+     * @param hash String
+     * @return ByteArray?
      */
-    fun fetchContract(contractId: String, cb: DapiRequestCallback<Contract>) {
-        val request = JsonRPCRequest("fetchContract",
-                mapOf("contractId" to contractId))
-        dapiService.fetchContract(request).enqueue({
-            val body = it.body()
-            if (it.isSuccessful && body != null) {
-                if (body.error != null) {
-                    cb.onError(body.error["message"] as String)
-                } else {
-                    val contract = ContractFactory().create(body.result!!.toMutableMap())
-                    cb.onSuccess(JsonRPCResponse(contract, body.id, body.jsonrpc, body.error))
-                }
-            } else {
-                cb.onError(it.message())
-            }
-        }, {
-            cb.onError(it.localizedMessage)
-        })
+
+    fun getBlockByHash(hash: String): ByteArray? {
+        Preconditions.checkArgument(hash.length == 64)
```

Comment:
What's `64`? Can we store it in a `val`?

Reply:
64 is the size of a 32 byte hash value as a Hex String.  This will be addressed.

Reply:
Resolved here: 2b52b25